### PR TITLE
Add prototypes for Property View and Ecore Model Editor

### PR DESCRIPTION
**Pull request by ndoschek**
- Restructure client repository

- Property View Prototype
  - Add first prototype for the diagram element property view
  - property-view dependencies (locally packaged npm packages) are temporarily stored in ext-packages
  - Add custom glsp selectionmodule (will be integrated in glsp itself later on)
  - Implement glsp selection data service: glsp selection provide semantic uris for graphical model elements
  - Provide ui schemas in dedicated directory in workspace
  - Turn off jetty debug logs for modelserver product
 
- Ecore Model Editor Prototype
  - Customize theia-tree-editor as Ecore Model Editor
  - Add gifs as tree icons (extracted from Eclipse Sirius repo)
  - Remark: Tree and form updating is not finished fully yet
    - Nested nodes not correctly updated
    - Add/remove nodes on update and via tree node buttons not fully finished
    - Form widget: theia-tree-editor does not provide asynchronous methods to be able to keep the data up-to-date

- Update diagram tool palette styling
  - Remove tweaked name editing in the SubscriptionListener
  - Remove tweaking in command executing for name editing in SubscriptionListener
  - Remark: Now the editing of names is not supported for property-view and ecore model editor anymore (name fields are disabled temporarily)

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

### CODE REVIEW

File: client/packages/ecore-property-view/src/browser/properties-service.ts [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+    async getUiSchema(selection: any, properties?: any): Promise<UISchemaElement | undefined> {
+        if (properties && properties.eClass) {
+            const eClassName = this.getSelectionData(selection).eClass || properties.eClass.split("#//")[1];
+            return this.modelServerClient.getUiSchema(eClassName.toLowerCase()).then((response: any) => response.body as UISchemaElement);
+        }
+        return undefined;
+    }
+
+    protected ecoreTypeSchema = {
```
Comment (CamilleLetavernier) [inline client/packages/ecore-property-view/src/browser/properties-service.ts:105]: Can we move this schema into its own *.json file?

Comment (CamilleLetavernier) [inline client/packages/ecore-property-view/src/browser/properties-service.ts:105]: Can we move this schema into its own *.json file?

File: client/packages/ecore-tree-editor/src/browser/style/ecore-tree-editor.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+.ecoreimg {
+    background-size: 16px 16px;
+    height: 16px;
+    width: 16px;
+    margin-right: 6px;
+}
+
+.eattribute {
+    background-image: url(../../../../sprotty-ecore/images/EAttribute.gif);
```
Comment (CamilleLetavernier) [inline client/packages/ecore-tree-editor/src/browser/style/ecore-tree-editor.css:37]: These long relative URIs feel a bit fragile. Can we improve this, e.g. by sharing resources into a different package?

File: client/packages/ecore-tree-editor/src/browser/tree/tree-model-service.ts [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+    getChildrenMapping(): Map<string, TreeEditor.ChildrenDescriptor[]> {
+        return EcoreModel.childrenMapping;
+    }
+
+    getNameForType(type: string): string {
+        return EcoreModel.Type.name(type);
+    }
+
+    public ecoreTypeSchema = {
```
Comment (CamilleLetavernier) [inline client/packages/ecore-tree-editor/src/browser/tree/tree-model-service.ts:98]: Can we move this schema to a separate file as well?

File: server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/EcoreFacade.java [hunk 4]
```diff
@@ -162,6 +163,12 @@ public void initializeNotationElement(NotationElement element, EObject semanticE
 		return diagram.getElements().stream().filter(element -> element.getSemanticElement() == null)
 				.collect(Collectors.toList());
 	}
+	
+	public List<NotationElement> findOldShapes(EClassifier eClassifier) {
```
Comment (CamilleLetavernier) [inline server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/EcoreFacade.java:167]: This method can be removed; we still need to implement a reliable way to uniquely identify model-server elements

**Review by CamilleLetavernier — COMMENTED** [review]
I've not tested the changes yet; the comments are only based on a first high-level code review. In general, I would move all UI/Schemas definitions to their own file

**Comment by CamilleLetavernier** [comment]
Can we change the Property Labels to match the case specified in the UI Schema, rather than always using upper case? Also, can the Checkboxes be left-aligned rather than centered?
image

I don't know if this is related to how EMF handles default values, but it seems that "0" is never presented in this integer-value field:

image

Values other than 0 work fine:

image

File: client/packages/ecore-tree-editor/src/browser/style/ecore-tree-editor.css (hunk 2 above)
Comment (ndoschek) [reply client/packages/ecore-tree-editor/src/browser/style/ecore-tree-editor.css:37]: That's a good idea! I moved them to a new common package and also moved the css classes there, as they are used by both the command palette and the tree editor.

File: client/packages/ecore-property-view/src/browser/properties-service.ts (hunk 1 above)
Comment (ndoschek) [reply client/packages/ecore-property-view/src/browser/properties-service.ts:105]: I moved it temporarily to the new common package, such that both the property view and the tree editor can use it. It is already available to fetch via the modelserver, but I noticed severe performance issues when the complete schema is fetched. Therefore we use temporarily this subset of the schema locally.

**Comment by ndoschek** [comment]
Thank you for your review Camille!
I addressed your inline comments and adjusted the styling of the property view. In that run, I aligned the styling of the tree editor and also removed the `uppercase` styling of the labels.

Regarding the `0` value:
EMF does not save the default value in that case and therefore the modelserver delivers a literal without any value property.
But displaying zero is also an issue in the version of JsonForms that is used at the moment.
I would suggest to keep track of that issue, and as soon as we consume the property view via the actual dependency (and JSONForms is updated too) we can fix that by initializing the value property for example.

**Review by CamilleLetavernier — APPROVED** [review]
It looks great! Thank you

**Comment by CamilleLetavernier** [comment]
> Regarding the `0` value:
> EMF does not save the default value in that case and therefore the modelserver delivers a literal without any value property.
> But displaying zero is also an issue in the version of JsonForms that is used at the moment.
> I would suggest to keep track of that issue, and as soon as we consume the property view via the actual dependency (and JSONForms is updated too) we can fix that by initializing the value property for example.

:+1:

**Pull request merged**